WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderViewModel.kt

```diff
@@ -447,7 +449,8 @@ class ReaderViewModel @Inject constructor(
     fun onSubFilterItemSelected(item: SubfilterListItem) {
         when (item) {
             is SubfilterListItem.SiteAll -> clearTopBarFilter()
-            is SubfilterListItem.Site -> updateTopBarFilter(item.blog.name, ReaderFilterType.BLOG)
+            is SubfilterListItem.Site -> updateTopBarFilter(item.blog.name
+                .ifEmpty { urlUtilsWrapper.removeScheme(item.blog.url.ifEmpty { "" }) }, ReaderFilterType.BLOG)
```

Comment:
❓ I see this logic of removing scheme is used here and also in `ReaderPostUiStateBuilder:436`. Would it make sense to extract this behavior into a common function (e.g. extension function or some utility class) to avoid duplicating it?

Reply:
Yes, that's correct. For now, I think it's fine to keep it separate. In the next version we could improve it, since there could be a bit of refactoring around displaying the correct name, and use it for filter and sorting.